BEST_PRACTICES.md

```diff
@@ -1,21 +1,105 @@
-# Pepr Best Practices (WIP)
+# Pepr Best Practices
 
 ## TOC
 
-- [Store](#pepr-store)
+- [Core Development](#core-development)
+- [Debugging](#debugging)
+- [Deployment](#deployment)
+- [Keep Modules Small](#keep-modules-small)
+- [Monitoring](#monitoring)
+- [Multiple Modules or Multiple Capabilities](#multiple-modules-or-multiple-capabilities)
 - [OnSchedule](#onschedule)
+- [Security](#security)
+- [Store](#pepr-store)
 - [Watch](#watch)
 
 
-## Pepr Store
 
-The store is backed by ETCD in a `PeprStore` resource, and updates happen at 5-second intervals when an array of patches is sent to the Kubernetes API Server. The store is intentionally not designed to be `transactional`; instead, it is built to be eventually consistent, meaning that the last operation within the interval will be persisted, potentially overwriting other operations. In simpler terms, changes to the data are made without a guarantee that they will occur simultaneously, so caution is needed in managing errors and ensuring consistency.  
+## Core Development
+
+When developing new features in Pepr Core, it is recommended to use `npx pepr deploy -i pepr:dev`, which will deploy Pepr's Kubernetes manifests to the cluster with the development image. This will allow you to test your changes without having to build a new image and push it to a registry. 
+
+The workflow for developing features in Pepr is:
+1. Run `npm test` which will create a k3d cluster and build a development image called `pepr:dev`
+2. Deploy development image into the cluster with `npx pepr deploy -i pepr:dev`
+
+## Debugging
+
+Pepr can be broken down into two parts: Admission and Watches. If the focus of the debug is on a Mutation or Validation, then only pay attention to pods with labels `pepr.dev/controller: admission`, else, you can focus on `pepr.dev/controller: watch`.
+
+## Deployment
+
+Production environment deployments should be `declarative` and `imperative` in order to avoid mistakes. The Pepr modules should be generated with `npx pepr build` and moved into the appropriate location.
```

Comment:
```suggestion
Production environment deployments should be `declarative` in order to avoid mistakes. The Pepr modules should be generated with `npx pepr build` and moved into the appropriate location.
```